### PR TITLE
Ladybird: Open target _blank links in new tab

### PR DESCRIPTION
**Pull request by Coderdreams**

### CODE REVIEW

File: Ladybird/WebContentView.cpp [hunk 1]
```diff
@@ -778,12 +778,10 @@ void WebContentView::notify_server_did_unhover_link(Badge<WebContentClient>)
 
 void WebContentView::notify_server_did_click_link(Badge<WebContentClient>, AK::URL const& url, DeprecatedString const& target, unsigned int modifiers)
```
Comment (konradekk) [inline Ladybird/WebContentView.cpp:779]: (click is not the best name, i guess—maybe `activated` or `followed`?)

File: Ladybird/WebContentView.cpp [hunk 2]
```diff
@@ -778,12 +778,10 @@ void WebContentView::notify_server_did_unhover_link(Badge<WebContentClient>)
...
 {
-    // FIXME
-    (void)url;
-    (void)target;
+    if (on_open_url_in_new_tab && target == "_blank") {
+        on_open_url_in_new_tab(url, Web::HTML::ActivateTab::Yes);
```
Comment (konradekk) [inline Ladybird/WebContentView.cpp:782]: care to add `TODO` for choosing whether tab should be activated or not? 😅 [e.g. through a browser setting])

**Review by konradekk — APPROVED** [review]
👍

File: Ladybird/WebContentView.h [hunk 3]
```diff
@@ -51,6 +51,7 @@ class WebContentView final
     virtual ~WebContentView() override;
 
     Function<String(Web::HTML::ActivateTab)> on_new_tab;
+    Function<String(const AK::URL&, Web::HTML::ActivateTab)> on_open_url_in_new_tab;
```
Comment (linusg) [inline Ladybird/WebContentView.h:54]: We already have `on_link_click`, please don't deviate from `Browser` when possible.

https://github.com/SerenityOS/serenity/blob/c048cf2004c9cda9ed6533b81b586600b740c14b/Userland/Applications/Browser/Tab.cpp#L265-L271

**Review by linusg — CHANGES_REQUESTED** [review]

File: Ladybird/WebContentView.h (hunk 3 above)
Comment (Coderdreams) [reply Ladybird/WebContentView.h:54]: I'd need to check what to do, it seems Ladybird already deviates a little from userland browser.

File: Ladybird/WebContentView.cpp (hunk 2 above)
Comment (Coderdreams) [reply Ladybird/WebContentView.cpp:782]: I'll see about that, I got another comment and maybe the code changes.

File: Ladybird/WebContentView.h (hunk 3 above)
Comment (Coderdreams) [reply Ladybird/WebContentView.h]: Please check now, the code looks a little more like serenityos's Browser but with normal current browsers' activate behavior, ie: middle click anything opens in an onfocused tab

Comment (linusg) [reply Ladybird/WebContentView.h]: Thanks!

**Pull request merged**